### PR TITLE
Presign options

### PR DESCRIPTION
**Pull request by sandeep-paliwal**
Support passing permissions to generate presign URL
supported permissions
READ, WRITE, DELETE  (Default READ)

## How Has This Been Tested?

unit tests and manual tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: lib/Files.js [hunk 1]
```diff
@@ -39,6 +39,12 @@ function throwNotImplemented (funcName) {
   logAndThrow(new codes.ERROR_NOT_IMPLEMENTED({ messageValues: [funcName] }))
 }
 
+const _permissions = {
```
Comment (moritzraho) [inline lib/Files.js:42]: Here you can define a static constant instead: `Files.Permissions = {}` that way it is exposed to the user too

File: lib/Files.js [hunk 2]
```diff
@@ -55,6 +61,7 @@ class Files {
    * @private
    */
   constructor (_isTest) {
+    this.permissions = _permissions
```
Comment (moritzraho) [inline lib/Files.js:64]: please remove this

File: lib/Files.js [hunk 3]
```diff
@@ -731,6 +738,7 @@ class Files {
    * @param {object} options Options to generate presign URL
    * @param {string} options.blobName file path
    * @param {number} options.expiryInSeconds presign URL expiry duration
+   * @param {string} options.permissions premissions for presigned URL
```
Comment (moritzraho) [inline lib/Files.js:741]: could you add a validateInput for the options like in https://github.com/adobe/aio-lib-files/blob/784b6fffe07b2db369702983664020a9443d8153/lib/Files.js#L430 for example ?

File: lib/impl/AzureBlobFiles.js [hunk 4]
```diff
@@ -348,6 +348,11 @@ class AzureBlobFiles extends Files {
     if (!options || !options.expiryInSeconds) {
       logAndThrow(new codes.ERROR_MISSING_OPTION({ messageValues: ['expiryInSeconds'], sdkDetails: [filePath, options] }))
     }
+
+    if (!options.permissions) {
+      options.permissions = this.permissions.READ
```
Comment (moritzraho) [inline lib/impl/AzureBlobFiles.js:353]: and here you can use Files.Permissions.Read instead of this

**Review by moritzraho — CHANGES_REQUESTED** [review]
2 minor comments in code.

Also could you regenerate the api doc + add the permission usage in the README ?

**Comment by sandeep-paliwal** [comment]
@purplecabbage I have verified your changes around FilePermissions export, looks Good.

**Review by purplecabbage — APPROVED** [review]

**Pull request merged**